src/sentry/utils/encryption.py

```diff
@@ -28,18 +28,22 @@ def encrypt(self, value):
         return b'{}${}'.format(self.default_scheme, scheme.encrypt(value))
 
     def decrypt(self, value):
+        # we assume that if encryption is not configured, it was never
+        # configured
+        if not self.schemes:
+            return value
         try:
             enc_method, enc_data = value.split('$', 1)
         except (ValueError, IndexError):
-            pass
-        else:
-            enc_data = smart_bytes(enc_data)
-            try:
-                scheme = self.schemes[enc_method]
-            except KeyError:
-                raise ValueError('Unknown encryption scheme: %s'.format(enc_method))
-            value = scheme.decrypt(enc_data)
-        return value
+            return value
```

Comment:
Even if that were true it wouldnt be synchronous so we'd have to have an intermittent phase.

tl;dr you should be able to enable encryption without downtime, and then backfill encrypted values